miniscope_io/stream_daq.py

```diff
@@ -177,7 +178,8 @@ def _trim(
         expected_payload_size = expected_size_array[0]
         expected_data_size = expected_size_array[header.frame_buffer_count]
 
-        if data.shape[0] != expected_payload_size:
+        # This is temporary. It's better to detect all dummy words and remove them.
+        if data.shape[0] != expected_payload_size + self.config.dummy_words * 4:
```

Comment:
To summarize, all of this dummy data should be cut earlier, just after preamble detection, as I mentioned in the above comment—one thing we'd want to refactor. That'll make the ordering stuff a bit more decoupled.

> shouldn't we revise the `buffer_npix` method

I thought this would be better too at first, but I did it this way because we need to trim based on the original definition and validate based on the new definition.

Reply:
Makes sense!